src/Xamarin.Android.Build.Tasks/Tasks/Javac.cs

```diff
@@ -57,16 +58,19 @@ protected override string GenerateCommandLineCommands ()
 
 			cmd.AppendSwitchIfNotNull ("-J-Dfile.encoding=", "UTF8");
 
-			cmd.AppendSwitchIfNotNull ("-d ", ClassesOutputDirectory);
-
-			cmd.AppendSwitchIfNotNull ("-classpath ", Jars == null || !Jars.Any () ? null : string.Join (Path.PathSeparator.ToString (), Jars.Select (i => i.ItemSpec)));
-			cmd.AppendSwitchIfNotNull ("-bootclasspath ", JavaPlatformJarPath);
-			cmd.AppendSwitchIfNotNull ("-encoding ", "UTF-8");
 			cmd.AppendFileNameIfNotNull (string.Format ("@{0}", TemporarySourceListFile));
 			cmd.AppendSwitchIfNotNull ("-target ", JavacTargetVersion);
 			cmd.AppendSwitchIfNotNull ("-source ", JavacSourceVersion);
 
 			return cmd.ToString ();
 		}
+
+		protected override void WriteOptionsToResponseFile (StreamWriter sw)
+		{
+			sw.WriteLine ($"-d {Files.GetShortPath (ClassesOutputDirectory).Replace (@"\", "\\")}");
+			sw.WriteLine ("-classpath \"{0}\"", Jars == null || !Jars.Any () ? null : string.Join (Path.PathSeparator.ToString (), Jars.Select (i => i.ItemSpec)));
```

Comment:
This should have the same "`\` is an escape character" issue as `-bootclasspath`, which you fixed earlier.